meos/src/general/set.c

```diff
@@ -985,8 +985,17 @@ set_make_free(Datum *values, int count, meosType basetype, bool order)
   assert(values); assert(count >= 0);
   Set *result = NULL;
   if (count > 0)
+  {
     result = set_make_exp(values, count, count, basetype, order);
-  pfree(values);
+#if MEOS
+    if (! basetype_byvalue(basetype))
+    {
+      for (int i = 0; i < count; i++)
+        pfree(DatumGetPointer(values[i]));
+    }
+#endif /* MEOS */
```

Comment:
We should not have an `#if MEOS` here. Either the function always frees the values or it never does, but having a different behaviour in MEOS and MobilityDB will be too confusing.